.github/workflows/test.yml

```diff
@@ -8,26 +8,50 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Don't we need macos-12 here?

Reply:
I'm going to have to reverify the test, looks like there's a problem with running freebsd because it was running on the "outer" Darwin OS instead of the freebsd VM.

As for the matrix, I had to cluster the freebsd one's (macos-12) under the `-include` list separately. I did try to use your formula `${{ matrix.os == 'linux' ? 'ubuntu-latest' : (matrix.os == 'macos' ? 'macos-latest' : ( matrix.os == 'windows' ? 'windows-latest' : 'macos-12')) }}`, but github complained about the syntax.

Reply:
Hm, I see...

I appreciate you for tackling this issue. Unfortunately, integrating the tests seems harder than expected and I don't want to bother you by fixing test files. If you find it hard to resolve, let's separate the FreeBSD test again like you did first. 

I will revisit this issue later.

Reply:
hmm... looks like freebsd-vm/vmactions is a single self-contained test step, ie it brings up the VM, performs `run` and then exits, causing the VM to be destroyed. https://github.com/vmactions/freebsd-vm/blob/main/index.js#L199

This is a problem for integrating FreeBSD with the other os targets...

Reply:
I've placed the freebsd matrix in its own block.
Results: https://github.com/ldnetgate/purego/actions/runs/5960355009